File: src/main/java/com/paragon/api/util/string/StringUtil.java
package com.paragon.api.util.string;

import java.util.Locale;

public final class StringUtil {

    public static String getFormattedText(final Enum<?> enumIn) {
        final String text = enumIn.name();
        final StringBuilder formatted = new StringBuilder(text.length());

        boolean isFirst = true;
        for (final char c : text.toCharArray()) {
            if (c == '_') {
                isFirst = true;
                continue;
            }

            if (isFirst) {
                formatted.append(String.valueOf(c).toUpperCase(Locale.ROOT));
                isFirst = false;
            } else {
                formatted.append(String.valueOf(c).toLowerCase(Locale.ROOT));
            }
        }

        return formatted.toString();
    }

    public static String wrap(final String str, final int length) {
        final StringBuilder result = new StringBuilder(str.length());
        int lastDelimPos = 0;

        for (final String token : str.split(" ", -1)) {
            if (result.length() - lastDelimPos + token.length() > length) {
                result.append(System.lineSeparator()).append(token);
                lastDelimPos = result.length() + 1;
            } else {
                result.append((result.length() == 0) ? "" : " ").append(token);
            }
        }

        return result.toString();
    }

}

File: src/main/kotlin/com/paragon/util/string/StringUtil.kt
package com.paragon.util.string

/**
 * @author Surge
 * @author SooStrator1136
 */
object StringUtil {

    @JvmStatic
    fun getFormattedText(enumIn: Enum<*>): String {
        val text = enumIn.name
        val formatted = StringBuilder(text.length)
        var isFirst = true
        for (c in text.toCharArray()) {
            if (c == '_') {
                isFirst = true
                continue
            }
            if (isFirst) {
                formatted.append(c.toString().uppercase())
                isFirst = false
            }
            else {
                formatted.append(c.toString().lowercase())
            }
        }
        return formatted.toString()
    }

    @JvmStatic
    fun wrap(str: String, length: Int): String {
        val result = StringBuilder(str.length)
        var lastDeliminatorPosition = 0

        for (token in str.split(" ")) {
            if (result.length - lastDeliminatorPosition + token.length > length) {
                result.append(System.lineSeparator()).append(token)
                lastDeliminatorPosition = result.length + 1
            }
            else {
                result.append(if (result.isEmpty()) "" else " ").append(token)
            }
        }

        return result.toString()
    }

}